### ISSUE TITLE
ConnectionResetError: [Errno 54] Connection reset by peer when running "process_images.py"

### ISSUE THREAD

**Issue by vinaysp** [YaleDHLab/pix-plot#56]
Hi all,

I am trying to run the "process_images.py" file and I am certain I have all the correct dependencies installed correctly but the following error messages appear. Please note that I did make a small change to the "process_images.py" file, by changing the number of clusters hence why file is called process_imagesv2.py and have not changed anything else.  That said I have no idea why this error is appearing and would appreciate any help to resolving this issue. 

Thanks.
<img width="768" alt="screen shot 2018-06-15 at 14 32 33" src="https://user-images.githubusercontent.com/32655933/41471138-aac397f4-70aa-11e8-967b-70d7b014b2e0.png">

**Comment by duhaime** [comment]
@vinaysp Do you have an internet connection? You'll need one to download the imagenet model if you don't already have it on your machine. If you do have an internet connection, I'd try again. If the problem persists, we may need to change the model url...

**Comment by vinaysp** [comment]
@duhaime Thanks for getting back to me. Yes, I have an internet connection. I tried running the same file again but the exact same error as shown above is still occurring. Is there anymore information I can provide you that could help in debugging this issue?

**Comment by pleonard212** [comment]
As a temporary workaround you could try to `wget http://download.tensorflow.org/models/image/imagenet/inception-2015-12-05.tgz` into `/tmp/imagenet` (or place it there via some other method).  If you untar/ungzip it the script should find it...

**Comment by vinaysp** [comment]
@pleonard212 I will give that a go and let you know what happens. Thanks.

**Comment by vinaysp** [comment]
@pleonard212 sorry for not getting back sooner, been a really busy couple of days for me. First of all should the tmp/imagenet directory already exist in my system? Apologies for the silly question. I also couldn't use the wget command so I downloaded the imagenet folder and created a tmp/imagenet directory within pix-plot-master folder. I then re-ran the script but the original problem still persists. Have you any other thoughts on where the root cause of the issue may lie?

Thanks.

**Comment by vinaysp** [comment]
@duhaime I tried this and I am able to download the imagenet folder (see first screenshot) but then I receive this other error.
<img width="1090" alt="screen shot 2018-06-20 at 18 40 33" src="https://user-images.githubusercontent.com/32655933/41675105-b501bdbe-74b9-11e8-8ff9-d4f57eddf666.png">
<img width="708" alt="screen shot 2018-06-20 at 18 41 14" src="https://user-images.githubusercontent.com/32655933/41675118-bb9f729c-74b9-11e8-8b87-f6957b6d48e0.png">

**Comment by duhaime** [comment]
@vinaysp excellent, we're making progress! It looks like you're only loading 20 images, which is unfortunately too few images to really work well with PixPlot. To make the most of this tool, you'll want to load some more images in -- 10000 or 20000 is a sweet spot.

If you're just testing things out, though, feel free to try 30 or so images and you should be all set. Can you try adding some more images then trying again?

**Comment by vinaysp** [comment]
@duhaime yes my initial thought was to just try and test out pixplot with a smaller dataset and once that works use around 10,000 images. Ok I will give try it out with 30 images and let you know how it goes. Thanks again for your help really appreciate it.

**Comment by duhaime** [comment]
@vinaysp Sure thing! I'd be grateful if you could follow up and let us know how everything goes with the slightly increased image set...

**Comment by vinaysp** [comment]
@duhaime I tried with the slightly larger image set and the script seems to be working as you would expect and even tried running the web server and looks like things are working as you would expect with images being plotted.
Thank you so much for your help, really appreciate it.

**Comment by duhaime** [comment]
@vinaysp Woohoo! It's my pleasure to assist. Is it alright to close this issue then?

**Comment by vinaysp** [comment]
@duhaime Yes I think the issue is resolved now so happy for it to be closed.

**Comment by duhaime** [comment]
Happy plotting!
